src/common.rs

```diff
@@ -33,6 +33,14 @@ pub enum Architecture {
     Xtensa,
 }
 
+/// A CPU sub-architecture.
+#[allow(missing_docs)]
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+#[non_exhaustive]
+pub enum SubArchitecture {
```

Comment:
I wonder if this should have a `None` variant instead of using `Option`. I'm not sure which is more ergonomic.

Reply:
Personally, I prefer to use `Option` - it makes it clear to anyone using the API that it is optional (without having to look through the enum values), and `Option` has a bunch of convenience APIs that may be useful.